pkg/networkservice/common/excludedprefixes/server_test.go

```diff
@@ -140,6 +122,34 @@ func TestUniqueRequestPrefixes(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func testWaitForFile(t *testing.T, directoryPath, fileName string) {
+	prefixes := []string{"172.16.1.0/24", "10.96.0.0/12"}
+
+	testConfig := strings.Join(append([]string{"prefixes:"}, prefixes...), "\n- ")
+	configPath := filepath.Join(directoryPath, fileName)
+	defer func() { _ = os.Remove(configPath) }()
+
+	chain := next.NewNetworkServiceServer(excludedprefixes.NewServer(context.Background(),
+		excludedprefixes.WithConfigPath(configPath)))
+	req := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Context: &networkservice.ConnectionContext{},
+		},
+	}
+
+	_, err := chain.Request(context.Background(), req)
+	require.NoError(t, err)
+	require.ElementsMatch(t, req.GetConnection().GetContext().GetIpContext().GetExcludedPrefixes(), []string{})
+
+	writingToFile(t, testConfig, configPath)
+
+	require.Eventually(t, func() bool {
+		_, err := chain.Request(context.Background(), req)
+		require.NoError(t, err)
+		return reflect.DeepEqual(req.GetConnection().GetContext().GetIpContext().GetExcludedPrefixes(), prefixes)
+	}, time.Second, time.Millisecond*100)
```

Comment:
Let's add a check that prefixes are empty when file is deleted

Reply:
Added.